### PR TITLE
ICU-22596 IBM-1388 converter data update

### PR DESCRIPTION
**Pull request by yumaoka**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22596
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by yumaoka** [comment]
There are some other ICU4J test data changes not related to the charset mapping data - te.res, testtypes.res and root.res. These files were updated when I generated ICU4J data from ICU4C tree. Eventually, these files should be updated later.

**Comment by yumaoka** [comment]
@markusicu 

> <UE78D> \xFC\x90 |4

Should we use `|1` (fallback from Unicode to code page)  instead of `|4` (good one way map from Unicode to code page)?

**Comment by markusicu** [comment]
> There are some other ICU4J test data changes not related to the charset mapping data - te.res, testtypes.res and root.res. These files were updated when I generated ICU4J data from ICU4C tree. Eventually, these files should be updated later.

I looked at the testdata source files and don't see anything that should be affected by modifying the IBM-1388 conversion table. I can't imagine that transliteration should be affected either. Please revert these three.

File: icu4c/source/data/mappings/ibm-1388_P100-2024.ucm [hunk 1]
```diff
@@ -20,6 +13,14 @@
 <icu:charsetFamily>           "EBCDIC"
 <icu:alias>                   "ibm-1388_VPUA"
 
+# includes mappings for surrogate pairs
+<icu:state>                   0-ff, e:1.s, f:0.s
+<icu:state>                   initial, 0-3f:4, e:1.s, f:0.s, 40:3, 41-fe:2, ff:4, cc:5, ce:5, f5:5, f7:5
+<icu:state>                   0-40:1.i, 41-fe:1., ff:1.i
+<icu:state>                   0-ff:1.i, 40:1.
+<icu:state>                   0-ff:1.i
+<icu:state>                   0-40:1.i, 41-fe:1.p, ff:1.i
```
Comment (markusicu) [inline icu4c/source/data/mappings/ibm-1388_P100-2024.ucm:22]: Why did the old table not need an explicit state table, but the new one does?
Did you try it without this explicit table?
I have a dim memory that genmbcs knows what to do with "EBCDIC_STATEFUL".

File: icu4c/source/data/mappings/ibm-1388_P100-2024.ucm [hunk 2]
```diff
@@ -1,16 +1,9 @@
 # Copyright (C) 2016 and later: Unicode, Inc. and others.
 # License & terms of use: http://www.unicode.org/copyright.html
-# *******************************************************************************
-# *
-# *   Copyright (C) 1995-2001, International Business Machines
-# *   Corporation and others.  All Rights Reserved.
-# *
-# *******************************************************************************
 #
-# File created by rptp2ucm (compiled on Dec 18 2001)
```
Comment (markusicu) [inline icu4c/source/data/mappings/ibm-1388_P100-2024.ucm:10]: Is the new version not created by rptp2ucm?

File: icu4c/source/data/mappings/ibm-1388_P100-2024.ucm [hunk 3]
```diff
@@ -32312,31 +32388,142 @@ CHARMAP
...
-<UE82B> \xCE\x6C |0
-<UE82C> \xCE\x6D |0
-<UE831> \xCE\x72 |0
-<UE832> \xCE\x73 |0
-<UE83B> \xCE\x7C |0
-<UE843> \xCE\x85 |0
-<UE854> \xCE\x96 |0
-<UE855> \xCE\x97 |0
-<UE864> \xCE\xA6 |0
+<UE78D> \xFC\x90 |4
```
Comment (markusicu) [inline icu4c/source/data/mappings/ibm-1388_P100-2024.ucm:32391]: You asked about |1 vs. |4 for these mappings. I think |4 is good. It "normalizes" (and preserves) old text that used PUA code points to the same byte sequence which now round-trips to a real character.

File: icu4c/source/data/mappings/ibm-1388_P100-2024.ucm (hunk 1 above)
Comment (yumaoka) [reply icu4c/source/data/mappings/ibm-1388_P100-2024.ucm:22]: My understanding is that it did not have surrogate pair mappings. And "EBCDIC_STATEFUL" is sufficient. I got the proposed change from Japan without state table and tried to build it. It ended up an error. After the investigation, it requires explicit state table when some codes mapped to supplemental planes.

File: icu4c/source/data/mappings/ibm-1388_P100-2024.ucm (hunk 2 above)
Comment (yumaoka) [reply icu4c/source/data/mappings/ibm-1388_P100-2024.ucm:10]: The group who managed CDRA is gone, and the data was migrated from Notes database to something else. My understanding is that it's no longer handle code pages in the way they previously did.

**Comment by yumaoka** [comment]
> I looked at the testdata source files and don't see anything that should be affected by modifying the IBM-1388 conversion table. I can't imagine that transliteration should be affected either. Please revert these three.

When build it, ICU4J data is not fully sync'ed with ICU4C. I also understand not all resources need to be affected, but the build script handle everything in one shot. Probably someone was lazy to update ICU4C, but did not update corresponding J data. I can rebase and try it again and see if there are any diffs.

File: icu4c/source/data/mappings/ibm-1388_P100-2024.ucm (hunk 2 above)
Comment (markusicu) [reply icu4c/source/data/mappings/ibm-1388_P100-2024.ucm:10]: Ok, but the following line is updated, referring to different versions of .RPMAP* & .TPMAP*.
It seems weird to just delete the first half of the sentence here.
Did someone run the tool, or did they fiddle manually with the mappings?

**Comment by markusicu** [comment]
> When build it, ICU4J data is not fully sync'ed with ICU4C. I also understand not all resources need to be affected, but the build script handle everything in one shot. Probably someone was lazy to update ICU4C, but did not update corresponding J data. I can rebase and try it again and see if there are any diffs.

If there are diffs, they should still be unrelated to the conversion table change. I think it's best not to change unrelated files in the same PR.

File: icu4c/source/data/mappings/ibm-1388_P100-2024.ucm (hunk 2 above)
Comment (yumaoka) [reply icu4c/source/data/mappings/ibm-1388_P100-2024.ucm:10]: Someone in GCoC Tokyo made the change - probably manually. I had to fix some issues on top of it (including state table).

**Review by markusicu — APPROVED** [review]

**Pull request merged**